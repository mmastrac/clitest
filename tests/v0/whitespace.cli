#!/usr/bin/env clitest --v0

$ printf "  X  \n"
!   X

$ printf "  X  \n"
?   X

$ printf "  X  \n"
? ^  X  $

$ printf "  X  \n"
!!!
  X
!!!

$ printf "  X  \n"
???
  X
???

$ printf "  X  \n"
???
^  X  $
???

$ printf "   X   \n"
%EXPECT_FAILURE
!   X

$ printf "   X   \n"
%EXPECT_FAILURE
?   X

$ printf "   X   \n"
%EXPECT_FAILURE
? ^  X  $

$ printf "abc\n\ndef\n"
# Indented !!! blocks strip indentation from their interior lines to match
# Note that the empty line is actually empty (no indent)
  !!!
  abc

  def
  !!!
