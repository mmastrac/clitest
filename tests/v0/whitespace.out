Running <root>/whitespace.cli ...
printf "  X  \n"
---
  X  
---
✅ OK

printf "  X  \n"
---
  X  
---
✅ OK

printf "  X  \n"
---
  X  
---
✅ OK

printf "  X  \n"
---
  X  
---
✅ OK

printf "  X  \n"
---
  X  
---
✅ OK

printf "  X  \n"
---
  X  
---
✅ OK

printf "   X   \n"
---
   X   
---
✅ OK

printf "   X   \n"
---
   X   
---
✅ OK

printf "   X   \n"
---
   X   
---
✅ OK

printf "abc\n\ndef\n"
---
abc

def
---
✅ OK

<root>/whitespace.cli PASSED
